packages/server/lib/controllers/sync.controller.integration.test.ts

```diff
@@ -0,0 +1,692 @@
+/* eslint-disable @typescript-eslint/unbound-method */
+import { describe, it, expect, vi, beforeEach } from 'vitest';
+import type { Request, Response, NextFunction } from 'express';
+import syncController from './sync.controller.js';
+import type { DBTeam, DBEnvironment, ConnectSession, DBUser, EndUser } from '@nangohq/types';
+import type { RequestLocals } from '../utils/express.js';
+import { syncManager } from '@nangohq/shared';
+
+const mockRunSyncCommand = vi.spyOn(syncManager, 'runSyncCommand').mockResolvedValue({
+    success: true,
+    response: true,
+    error: null
+});
+
+const createMockRequest = ({ body = {}, query = {}, params = {}, headers = {} as Record<string, any> } = {}): Request => {
+    return {
+        body,
+        params,
+        query,
+        get: (header: string) => headers[header]
+    } as Request;
+};
+const createMockResponse = (): Response<any, Required<RequestLocals>> => {
+    return {
+        status: vi.fn().mockReturnThis(),
+        send: vi.fn().mockReturnThis(),
+        locals: {
+            authType: 'secretKey',
+            account: { id: 0 } as DBTeam,
+            environment: { id: 1 } as DBEnvironment,
+            user: { id: 0 } as DBUser,
+            connectSession: { id: 0 } as ConnectSession,
+            endUser: { id: 0 } as EndUser
+        }
+    } as unknown as Response<any, Required<RequestLocals>>;
+};
+
```

Comment:
tests are not exhaustive, I have mostly added tests for parameters validation but it is better than zero tests. :/